filters/include/pcl/filters/impl/normal_space.hpp

```diff
@@ -217,7 +216,7 @@ pcl::NormalSpaceSampling<PointT, NormalT>::applyFilter (std::vector<int> &indice
       // Picking up a sample at random from jth bin
       do
       {
-        random_index = static_cast<unsigned int> ((*rng_uniform_distribution_) () % M);
+        random_index = rng_uniform_distribution_ () % M;
```

Comment:
Prefer changing the parameters instead:
* [permanently](https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution/param)
* [temporarily](https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution/operator())
since modulus operation doesn't preserve random-ness